File: benches/mini_bench.py
import time
import regex

text = "The quick brown fox jumps over the lazy dog."
long_text = text * 100

print("mrab-regex benchmarks (us/iter)")
print("================================")

# Short text
re1 = regex.compile(r"(?:quikc){e<=1}")
start = time.perf_counter()
for _ in range(1000): re1.search(text)
print(f"short text, 1 edit:    {(time.perf_counter()-start)*1000:>8.1f}")

re2 = regex.compile(r"(?:qwick){e<=2}")
start = time.perf_counter()
for _ in range(1000): re2.search(text)
print(f"short text, 2 edits:   {(time.perf_counter()-start)*1000:>8.1f}")

# Long text
re3 = regex.compile(r"(?:lorem){e<=2}", regex.IGNORECASE)
start = time.perf_counter()
for _ in range(10): re3.search(long_text)
print(f"long text, 2 edits:    {(time.perf_counter()-start)*100000:>8.1f}")

re4 = regex.compile(r"(?:xyzzy){e<=1}")
start = time.perf_counter()
for _ in range(10): re4.search(long_text)
print(f"long text, no match:   {(time.perf_counter()-start)*100000:>8.1f}")

File: benches/mini_bench.rs
use fuzzy_regex::FuzzyRegex;
use std::time::Instant;

fn main() {
    let text = "The quick brown fox jumps over the lazy dog.";
    let long_text = text.repeat(100);

    println!("fuzzy-regex benchmarks (us/iter)");
    println!("================================");

    // Short text
    let re = FuzzyRegex::new("(?:quikc){e<=1}").unwrap();
    let start = Instant::now();
    for _ in 0..1000 {
        let _ = re.find(text);
    }
    println!(
        "short text, 1 edit:    {:>8.1}",
        start.elapsed().as_micros() as f64 / 1000.0
    );

    let re = FuzzyRegex::new("(?:qwick){e<=2}").unwrap();
    let start = Instant::now();
    for _ in 0..1000 {
        let _ = re.find(text);
    }
    println!(
        "short text, 2 edits:   {:>8.1}",
        start.elapsed().as_micros() as f64 / 1000.0
    );

    // Long text
    let re = FuzzyRegex::new("(?:lorem){e<=2}").unwrap();
    let start = Instant::now();
    for _ in 0..10 {
        let _ = re.find(&long_text);
    }
    println!(
        "long text, 2 edits:    {:>8.1}",
        start.elapsed().as_micros() as f64 / 10.0
    );

    let re = FuzzyRegex::new("(?:xyzzy){e<=1}").unwrap();
    let start = Instant::now();
    for _ in 0..10 {
        let _ = re.find(&long_text);
    }
    println!(
        "long text, no match:   {:>8.1}",
        start.elapsed().as_micros() as f64 / 10.0
    );
}
